mesonbuild/build.py

```diff
@@ -449,6 +451,9 @@ def process_kwargs(self, kwargs, environment):
         if not isinstance(valalist, list):
             valalist = [valalist]
         self.add_compiler_args('vala', valalist)
+        if not isinstance(self, Executable):
+            self.vala_header = kwargs.get('vala_header', None)
+            self.vala_vapi = kwargs.get('vala_vapi', None)
```

Comment:
Why not use `kwargs.get('vala_header', self.name + '.h')`, etc here? Simplifies the code a bit.